tasks/updateVamms.ts

```diff
@@ -0,0 +1,132 @@
+
+import { task } from "hardhat/config";
+import mustache from "mustache";
+import path from "path";
+import { getConfig } from "../deployConfig/config";
+import { getPool } from "../poolConfigs/pool-addresses/pools";
+import { getPoolConfig } from "../poolConfigs/pool-configs/poolConfig";
+import { getSigner } from "./utils/getSigner";
+import { IVAMM } from "../typechain";
+
+interface SingleUpdateTemplateData {
+  vammAddress: string;
+
+  updateMaturityBuffer: boolean;
+  maturityBuffer: BigNumberish;
+
+  updateFeeProtocol: boolean;
+  feeProtocol: BigNumberish;
+
+  updateFee: boolean;
+  fee: BigNumberish;
+}
+
+interface MultisigTemplateData {
+  factoryAddress: string;
+  multisig: string;
+  chainId: string;
+
+  pools: SingleUpdateTemplateData[];
+}
+
+function writeGnosisSafeTemplate(data: MultisigTemplateData) {
+  // Get external template with fetch
+  const fs = require("fs");
+  const template = fs.readFileSync(
+    path.join(__dirname, "templates/updateVamms.json.mustache"),
+    "utf8"
+  );
+  const output = mustache.render(template, data);
+
+  fs.writeFileSync(
+    path.join(__dirname, "JSONs/updateVamms.json"),
+    output,
+    () => {}
+  );
+}
+
+// Description:
+//   This task generates multisig txs to update configuration of vamm.
+//     - First, you need to update the configuration in './poolConfigs/pool-configs/poolConfig.ts'
+//   for the corresponding pools
+//     - Then, run this task with the flags set depending what parameters you like to update
+//
+// Example:
+//   ``npx hardhat updateVamms --network mainnet aUSDC_v11 aDAI_v4 --maturity-buffer --fee-protocol --fee``
+
+task("updateVamms", "It updates the configurations of given pools")
+  .addVariadicPositionalParam("pools", "Space-separated pool names")
+  .addFlag("maturityBuffer")
+  .addFlag("feeProtocol")
+  .addFlag("fee")
+  .addFlag("multisig")
+  .addOptionalParam("underlyingNetwork", "The underlying network of the fork")
+  .setAction(async (taskArgs, hre) => {
+    // Fetch pool details
+    const poolNames: string[] = taskArgs.pools;
+
+    // Fetch factory
+    const factory = await hre.ethers.getContract("Factory");
+
+    // Retrieve multisig address for the current network
+    const network: string = taskArgs.underlyingNetwork || hre.network.name;
+    const deployConfig = getConfig(network);
+    const multisig = deployConfig.multisig;
+
+    // Initialize the data keeper
+    const data: MultisigTemplateData = {
+      pools: [],
+      factoryAddress: factory.address,
+      multisig: multisig,
+      chainId: await hre.getChainId(),
+    };
+
+    for (const pool of poolNames) {
+      const poolConfig = getPoolConfig(network, pool);
+      const poolDetails = getPool(network, pool);
+
+      const singleData: SingleUpdateTemplateData = {
+        vammAddress: poolDetails.vamm,
+
+        updateMaturityBuffer: taskArgs.maturityBuffer ? true : false,
+        maturityBuffer: poolConfig.maturityBuffer,
```

Comment:
pls let's double check if the Maturity buffer is in WAD (pool config vs contract)